AWSPinpoint/AWSPinpointEndpointProfile.m

```diff
@@ -520,15 +525,27 @@ - (void)encodeWithCoder:(NSCoder *)encoder {
 
 @implementation AWSPinpointEndpointProfileUser
 
-- (NSString*) quotedString:(NSString*) input {
+- (NSString *)quotedString:(NSString*) input {
     return [NSString stringWithFormat:@"\"%@\"", input];
 }
 
-- (NSString*) description {
+- (NSString *)description {
+    if (!self.userAttributes) {
```

Comment:
Nil `userAttributes` is going to fail the [unit test](https://github.com/aws-amplify/aws-sdk-ios/blob/main/AWSPinpointUnitTests/AWSPinpointNSSecureCodingTests.m):

Reply:
That's interesting that this is failing for you... it is not failing for me.  When `self.userAttributes =nil`, i'm seeing this function return a `NSString` with a value of something like:
```
....."OptOut" : "ALL","Attributes":{},"Metrics":{}, "User":{"UserId" : "(null)","Attributes":(null)}}
```